Cubical/HITs/Join/Properties.agda

```diff
@@ -0,0 +1,274 @@
+{-
+
+This file contains:
+
+- Equivalence with the pushout definition
+  Written by: Loïc Pujet, September 2019
+
+- Associativity of the join
+  Written by: Loïc Pujet, September 2019
+
+-}
+
+{-# OPTIONS --cubical --safe #-}
+
+module Cubical.HITs.Join.Properties where
+
+open import Cubical.Core.Glue
+
+open import Cubical.Foundations.Prelude
+open import Cubical.Foundations.Equiv
+open import Cubical.Foundations.Isomorphism
+open import Cubical.Foundations.GroupoidLaws
+
+open import Cubical.Data.Prod
+
+open import Cubical.HITs.Join.Base
+open import Cubical.HITs.Pushout
+
+private
+  variable
+    ℓ : Level
+    ℓ' : Level
+
+
+-- Alternative definition of the join using a pushout
+joinAlt : (A : Type ℓ) → (B : Type ℓ') → Type (ℓ-max ℓ ℓ')
+joinAlt A B = Pushout {_} {_} {_} {A × B} {A} {B} proj₁ proj₂
+
+-- Proof that it is equal
+joinAlt-iso-join : (A : Type ℓ) → (B : Type ℓ') → Iso (joinAlt A B) (join A B)
+joinAlt-iso-join A B = iso (joinAlt→join A B) (join→joinAlt A B) (join→joinAlt→join A B) (joinAlt→join→joinAlt A B)
+  where
+    joinAlt→join : (A : Type ℓ) → (B : Type ℓ') → joinAlt A B → join A B
```

Comment:
Can't you just remove A and B from this and the other helper functions? They are bound in the top-level function.

Reply:
Indeed !
Btw, I am becoming less and less of a fan of the A→B, B→A, A→B→A, B→A→B naming scheme for helper functions, because it is annoyingly long and already explicit in the type. 
Since they should probably only exist during the definition of an isomorphism, maybe a simpler naming scheme like f, inv, sec, ret should be adopted ?

Reply:
I can never remember which of section and retraction is which... But sure, you can use other names for these things. They are local anyway.